src/librustdoc/html/highlight.rs

```diff
@@ -57,20 +81,51 @@ fn write_header(out: &mut Buffer, class: Option<&str>, extra_content: Option<Buf
     }
 }
 
-fn write_code(out: &mut Buffer, src: &str, edition: Edition) {
-    // This replace allows to fix how the code source with DOS backline characters is displayed.
-    let src = src.replace("\r\n", "\n");
-    Classifier::new(&src, edition).highlight(&mut |highlight| {
-        match highlight {
-            Highlight::Token { text, class } => string(out, Escape(text), class),
-            Highlight::EnterSpan { class } => enter_span(out, class),
-            Highlight::ExitSpan => exit_span(out),
+fn write_code<'a>(out: &mut Buffer, src: impl Iterator<Item = Line<'a>>, edition: Edition) -> bool {
+    let mut iter = src.peekable();
+    let mut expand = false;
+
+    // For each `Line`, we replace DOS backlines with '\n'. This replace allows to fix how the code
+    // source with DOS backline characters is displayed.
+    while let Some(line) = iter.next() {
+        let (before, text, after) = match line {
+            Line::Hidden(text) => {
+                expand = true;
+                ("<span class=\"hidden\">", text.replace("\r\n", "\n"), "</span>")
```

Comment:
probably use a more specific class like hidden-code

Reply:
Good idea! That'll make it simpler for the background too.